File: src/main/java/com/aitho/Models/Teacher.java
package com.aitho.Models;

import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;
import org.springframework.data.mongodb.core.mapping.Field;

@Data
@Document(collection = "Teachers")
public class Teacher extends Person {

    @Id
    private String id;

    public Teacher(String name,String surname,String email,String password,String token) {
        super(name,surname,email,Role.Teacher,password,token);
    }
}

File: src/main/resources/front-end/angularFE/src/app/interface/teacher.ts
export interface Teacher {    
    name: string,
    surname: string,
    email: string,
    password: string,
    role: string,
    token: string
}

export interface TeacherRes {    
    name: string,
    surname: string,
    email: string,
}


export interface MappedTeacher {
    [id:string]: TeacherRes,
}


